src/OpenApi/src/Extensions/OpenApiEndpointRouteBuilderExtensions.cs

```diff
@@ -49,8 +49,16 @@ public static IEndpointConventionBuilder MapOpenApi(this IEndpointRouteBuilder e
                     using var writer = Utf8BufferTextWriter.Get(output);
                     try
                     {
-                        document.Serialize(new OpenApiJsonWriter(writer), documentOptions.OpenApiVersion);
-                        context.Response.ContentType = "application/json;charset=utf-8";
+                        if (UseYaml(pattern))
+                        {
+                            document.Serialize(new OpenApiYamlWriter(writer), documentOptions.OpenApiVersion);
+                            context.Response.ContentType = "application/yaml;charset=utf-8";
+                        }
+                        else
+                        {
+                            document.Serialize(new OpenApiJsonWriter(writer), documentOptions.OpenApiVersion);
```

Comment:
Are people going to be surprised when `openapi.toml` returns JSON, rather than failing?

Reply:
Maybe give more control for the user, passing all values in params, and give some magic extension .json or .yaml in the end of path if exists or not or wrong will be removed and set the right extension, for content type / mime type pass in the last param like mimeType: <default value> | default value like pattern: "openapi/v1" format: JSON mimeType: "application/json" \( 0-0)/

Reply:
> Are people going to be surprised when openapi.toml returns JSON, rather than failing?

Depends on user expectation and the type of experience that we want to provide. TOML isn't a supported OpenAPI document format so we could decide to throw if we encounter an extension _and_ it's not associated with the supported set of formats.

- Route with extension ending in `.yaml` or `yml` produces an OpenAPI document in YAML format.
- Route with extension ending in `.json` produces an OpenAPI document in JSON format.
- Route with extension not ending in any of the above produces an error.
- Route with no extension produces an OpenAPI document in JSON format.

With this behavior, #3 would be a breaking change between .NET 9 and .NET 10 but I think we can live with it.

> Maybe give more control for the user, passing all values in params, and give some magic extension .json or .yaml in the end of path if exists or not or wrong will be removed and set the right extension, for content type / mime type pass in the last param like mimeType: | default value like pattern: "openapi/v1" format: JSON mimeType: "application/json" ( 0-0)/

Yeah, I considered this option as well but opted to be more conservative in the amount of new API that was introduced. It also feels that given there are only two formats supported currently, we might want to support a simpler API for this.

Reply:
I've opted to avoid the breaking change for now. We've got runaway to revisit this if needed.